### ISSUE TITLE
What is the minimum number of data for single class to train model  from scratch at begining?

### ISSUE THREAD

**Issue by TapendraBaduwal** [ultralytics/ultralytics#2438]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I have train single class model from scratch with 10 image...but my model didi not detect ...Minimum amt of data for single class required is ?

**Comment by ExtReMLapin** [comment]
You'll need more than 10 instances lol

**Comment by TapendraBaduwal** [comment]
@ExtReMLapin Test on 16 images still accuracy not improve

log is here 

optimizer: SGD(lr=0.001) with parameter groups 57 weight(decay=0.0), 64 weight(decay=0.0005), 63 bias
train: Scanning /home/tapendra/Desktop/BrainTip VI86/datasets/train/labels.cache... 16 images, 0 backgrounds, 0 co
val: Scanning /home/tapendra/Desktop/BrainTip VI86/datasets/valid/labels.cache... 16 images, 0 backgrounds, 0 corr
Plotting labels to runs/detect/train/labels.jpg... 
Image sizes 640 train, 640 val
Using 0 dataloader workers
Logging results to runs/detect/train
Starting training for 10 epochs...

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
       1/10         0G      3.632      3.483      4.247         18        640: 100%|██████████| 8/8 [00:11<00:00, 
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 4/4 [00
                   all         16         64    0.00765      0.516      0.019    0.00637

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
       2/10         0G      3.358       3.47      4.272         10        640: 100%|██████████| 8/8 [00:08<00:00, 
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 4/4 [00
                   all         16         64    0.00739        0.5     0.0239    0.00687

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
       3/10         0G      3.593      3.428      4.209         25        640: 100%|██████████| 8/8 [00:09<00:00, 
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 4/4 [00
                   all         16         64    0.00758      0.516     0.0265    0.00735

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
       4/10         0G      3.315      3.302      4.234          8        640: 100%|██████████| 8/8 [00:09<00:00, 
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95):  75%|███████▌  | 3/4 [00

# Load a model
model = YOLO("yolov8n.yaml")  # build a new model from scratch
model.train(data="data.yaml", epochs=3)  # train the model

**Comment by ExtReMLapin** [comment]
Try with 100 images and 100 epochs.

**Comment by glenn-jocher** [comment]
@TapendraBaduwal hi, thank you for using YOLOv8! 

For object detection models, it is generally recommended to have a minimum of a few hundred annotated objects per class. With only 10 annotated images, your model may not have enough diversity in the learned features to sufficiently recognize objects in new images.

As a starting point, we suggest using at least 100 annotated images for a single class object detection model and train for more epochs (around 100) to potentially see improvements in accuracy. Please keep in mind that accuracy also depends on factors such as the quality of annotations and the complexity of the object(s) being detected.

Let us know if you have any other questions or concerns.

**Comment by NoufAlruwais** [comment]
Hi Glenn,

Thank you for your guidance on using YOLOv8 for object detection. I understand the recommendation to have a minimum of a few hundred annotated objects per class and to start with at least 100 annotated images for a single class, especially when training for more epochs to improve accuracy.

I'm curious to know, what are your recommendations or best practices for segmentation models? Specifically, how does the data requirement differ, and what should I consider in terms of the number of annotated images and training strategies?

Looking forward to your insights!

**Comment by glenn-jocher** [comment]
For segmentation models, the data requirements are generally higher due to the complexity of pixel-wise annotations. We recommend starting with at least 200-300 annotated images per class. Ensure high-quality annotations and consider using data augmentation techniques to enhance model performance. Training for more epochs and monitoring metrics like IoU can also help improve accuracy.

**Comment by NoufAlruwais** [comment]
Thanks for your advice on segmentation models. Appreciate your guidance!

**Comment by glenn-jocher** [comment]
@NoufAlruwais you're welcome! If you have any more questions or need further assistance, feel free to ask. Happy training!
